### PR TITLE
storage_file: boost url compute

### PR DESCRIPTION
**Pull request by simahawk**
URL is always computed because some parts of it might change.

Reading backend data that might come form server env params
or computing all the times the slug using regex can be very consuming.

Certain bits can be stored on their respective records.

### CODE REVIEW

**Comment by simahawk** [comment]
One more thing to consider: "served_by", "base_url", "directory_path", "url_include_directory_path" come from the env.
Does it make sense to have `@api.depends` on them?
This will still trigger the call to serve env machinery and, when served from odoo, the call to get_param.

**Comment by simahawk** [comment]
last commit is a blind porting and improvement of #85

File: storage_file/models/ir_actions_report.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import models
+
+
+class IrActionsReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    def render_qweb_pdf(self, res_ids=None, data=None):
+        return super(
+            IrActionsReport, self.with_context(print_report_pdf=True)
```
Comment (simahawk) [inline storage_file/models/ir_actions_report.py:13]: this key should be probably proposed to odoo core?

**Review by lmignon — APPROVED** [review]
LGTM (Code review)

**Comment by simahawk** [comment]
@JordiBForgeFlow good for you?

**Review by i-vyshnevska — APPROVED** [review]

**Pull request merged**